### PR TITLE
fix: Controlplane spec.class improvements

### PR DESCRIPTION
**Pull request by lsviben**
### Description of your changes

Add a better description for control plane spec.class and updates the classes gate flag to the current proposed one in https://github.com/upbound/spaces/pull/1916/files#r1805980906

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- ~[ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.~

### How has this code been tested
N/A

### CODE REVIEW

**Review by turkenh — APPROVED** [review]

File: apis/spaces/v1beta1/controlplane_types.go [hunk 1]
```diff
@@ -197,8 +197,11 @@ type ControlPlaneSpec struct {
 	// +kubebuilder:validation:XValidation:rule="!has(oldSelf.finishedAt) || oldSelf.finishedAt == self.finishedAt",message="finishedAt is immutable once set"
 	Restore *Restore `json:"restore,omitempty"`
 
-	// [[GATE:EnableClasses]]
-	// Class specifies the class of the control plane.
+	// [[GATE:EnableControlPlaneClasses]]
+	// Class specifies the class of the control plane. This affects the control plane sizing,
+	// including component replicas and resource requirements. Empty or default value
```
Comment (sttts) [inline apis/spaces/v1beta1/controlplane_types.go:202]:
```suggestion
	// including component replicas and resource requirements. Empty or "default"
```

File: apis/spaces/v1beta1/controlplane_types.go [hunk 2]
```diff
@@ -197,8 +197,11 @@ type ControlPlaneSpec struct {
...
+	// indicates a standard Spaces control plane without any additional class configuration.
+	// Valid classes need to be defined in the `class-config` configmap.
```
Comment (sttts) [inline apis/spaces/v1beta1/controlplane_types.go:204]: That was mostly the sentence I was missing.

File: apis/spaces/v1beta1/controlplane_types.go (hunk 1 above)
Comment (sttts) [reply apis/spaces/v1beta1/controlplane_types.go:202]: Maybe better: `This defaults to the value "default", meaning a standard ....`

**Review by sttts — APPROVED** [review]

File: apis/spaces/v1beta1/controlplane_types.go (hunk 2 above)
Comment (lsviben) [reply apis/spaces/v1beta1/controlplane_types.go:204]: We wont document this anywhere, unless we will at one point release this as a Spaces feature outside of Cloud

Hmm, maybe we should drop this info here. I can instead say: "Ask your Spaces admin for valid classes" The fact that this needs to be only for Cloud, but hidden from everybody else is making my life difficult 😄 .

**Pull request merged**